internal/run/run.go

```diff
@@ -141,45 +137,52 @@ func (w *Workflow) RunInner(ctx context.Context) error {
 
 		for t := range w.workflow.Targets {
 			sourceRes, targetRes, err := w.runTarget(ctx, t)
+			if sourceRes != nil {
+				w.SourceResults[sourceRes.Source] = sourceRes
+			}
+			w.TargetResults[t] = targetRes
+
```

Comment:
I think this causes a nil pointer 

See https://app.usepylon.com/issues?conversationID=9f64f2ba-0420-4ee9-acc9-629e10958b4d